### ISSUE TITLE
Node: Passing the wrong type to `rule.extract` causes a segmentation fault

### ISSUE THREAD

**Issue by thomastrapp** [html-extract/hext#24]
```js
var hext = require('hext');

var rule = new hext.Rule('<a href:link/>');

// rule.extract expects an object of type hext.Html
var result = rule.extract({}); // raises SIGSEGV
```

**Comment by thomastrapp** [comment]
This issue seems to be present in many node addons that use [Nan](https://github.com/nodejs/nan).

Nan's official documentation for [object wrapping](https://github.com/nodejs/nan/blob/f9e63d6924bc562ddea81ba941460da71efcfd24/doc/object_wrappers.md#passing-wrapped-objects-around) does not check the type and has the same error:

https://github.com/nodejs/nan/blob/f9e63d6924bc562ddea81ba941460da71efcfd24/doc/object_wrappers.md?plain=1#L231-L236

The [N-API](https://nodejs.org/api/n-api.html) has features that solve this specific problem with ["type-tagging"](https://nodejs.org/docs/latest-v17.x/api/n-api.html#object-wrap).

Dropping Nan in favor of the newer N-API might be a good idea.

**Comment by thomastrapp** [comment]
Fixed with commit 9aea4555d88113a7610d840fe61abcba5f5b5dd6
